Library/Homebrew/hardware.rb

```diff
@@ -39,14 +71,17 @@ def family
       end
 
       def cores
-        1
+        return @cores if @cores
+        @cores = Utils.popen_read("getconf", "_NPROCESSORS_ONLN").chomp.to_i
+        @cores = 1 unless $CHILD_STATUS.success?
+        @cores
       end
 
       def bits
-        case RUBY_PLATFORM
-        when /x86_64/, /ppc64/ then 64
-        when /i\d86/, /ppc/ then 32
-        end
+        return @bits if @bits
+        @bits = Utils.popen_read("getconf", "LONG_BIT").chomp.to_i
```

Comment:
This is new code, no?

Reply:
New to Homebrew, yes.

Reply:
Can the previous approach of using `RUBY_PLATFORM` be used here instead and e.g. fixed to work on ARM instead?

Reply:
`getconf LONG_BIT` works on both macOS and Linux, and is a POSIX standard. Using it would be my preference. If not, yes, I can adapt the previous approach for ARM.